### ISSUE TITLE
jQuery 1.11 Promises aren't Promises/A+ / ES6 compliant

### ISSUE THREAD

**Issue by blaggacao** [odoo/odoo#25725]
This is usually not troublesome in common odoo frontend tasks (which partly disqualifies this issue report, I know), but puts interoperability at stake (which is neither extremely highly rated, I know).

Nevertheless, I want to share with interested parties what's wrong with jQuery before 3.0:

- https://blog.domenic.me/youre-missing-the-point-of-promises/
- https://promisesaplus.com/implementations (search `jQuery`)

Which, under going concern assumptions, at some point makes migrating recommendable. Fortunately, there are tools that help with that:
- https://blog.jquery.com/2016/05/19/jquery-migrate-1-4-1-released-and-the-path-to-jquery-3-0/

/cc @sle-odoo

**Comment by blaggacao** [comment]
Today, I guess, I actually was stuck with this half a day as a `.then` chain would not execute in a predictable order. I wrapped it in a `$.when()`, instead...

**Comment by Yenthe666** [comment]
@blaggacao @sle-odoo barely has anything to do with jQuery / the JS framework. You need @JKE-be or @ged-odoo for that.
On a sidenote: I'm going to take a guess here that you'll get a response it is not a very huge problem and not considered a bug 😉

**Comment by blaggacao** [comment]
@Yenthe666 Hahaha, I can most certainly anticipate that...

No problem, folks. I know you have more important stuff to do.
Just in case, you weren't aware already...
Consider it a friendly ping.

**Comment by ged-odoo** [comment]
@blaggacao 

We actually talked  about moving from the jquery promises to the native promises.  For various reasons:
- we tried upgrading jquery from 1.11 to 3.x, and the jquery promises changed their semantics,
  which broke a few things
- performance (jquery promises are heavy)
- better semantics (this is the issue you raised in this ticket)
- reduce our dependency on jQuery 

But moving from jquery to native is a large task.  Upgrading jquery is probably much simpler, but we had too many small issues, and we had other more important bugs.

Anyway, this is something that I would like to do someday, but I have no idea if this would add enough value (relatively to other problems).  Clearly not a priority for now :/

Closing this ticket for now, but know that this is still something that we may do someday
